mypy/fastparse.py

```diff
@@ -279,6 +279,16 @@ def parse(
         # Disable deprecation warnings about \u
         with warnings.catch_warnings():
             warnings.filterwarnings("ignore", category=DeprecationWarning)
+            if options.allow_commented_type_annotations:
+                lines = str(source).split("\n")
+                result = ""
+                for line in lines:
+                    if "# type:" not in str(line):
+                        result += str(line) + str("\n")
+                    else:
+                        line = line.split("#")[0] + "\n"
+                        result += line
+                    source = result
```

Comment:
I think that this can also remove type comments within string literals, which is incorrect. Only type comments outside string literals (including triple-quoted ones) should be removed. Note that the `tokenize` stdlib module would probably do the trick, but it may be too slow to be used here, since it's implemented in Python.